### PR TITLE
ui/map: better rounding for distances

### LINKED ISSUE #29132: ui/map: better instruction distances
- [ ] round instead of floor - https://github.com/commaai/openpilot/pull/29333
- [ ] more improvements - https://github.com/commaai/openpilot/pull/29151
- [x] show "now" instead of 0 ft? - that's fine, GMaps does this as well. rounding should be good enough

### PR DESCRIPTION
**Pull request by deanlee**
rounding to nearest integer  if d <= 500 `ft | m`:
```
1.6 ->  2 ft | m
1.2 -> 1 ft | m
200.5 -> 201 ft | m
```
rounding by 2 digit precision  if d > 500 `ft | m` && d <= 10 `mi | km:`
```
0.52 -> 0.5 mi | km
0.89 -> 0.9 mi | km
1.52 -> 1.5 mi | km
1.58 -> 1.6 mi | km
```

rounding to nearest integer if d > 10 `mi | km`:
```
10.6 -> 11 mi | km
10.3 -> 10 mi | km
110.6 -> 111 mi | km

```

resolve https://github.com/commaai/openpilot/issues/29132

### CODE REVIEW

**Comment by incognitojam** [comment]
Nice! Can we use the same formatting for the ETA?
image

**Comment by sshane** [comment]
Think rounding to nearest 50 ft is fine, that's what GMaps does. But I haven't tried this in the car, so maybe it feels better

**Comment by sshane** [comment]
Let's still round to 50 ft/m, and continue to show ft/m < 500.

**Comment by sshane** [comment]
looks mostly good to me! @incognitojam can you final review it/clean it up if needed and get this merged?

File: selfdrive/ui/qt/maps/map_helpers.cc [hunk 1]
```diff
@@ -129,6 +129,23 @@ std::optional<QMapbox::Coordinate> coordinate_from_param(const std::string &para
...
+  };
+
+  d = std::max(d, 0.0f);
+  if (is_metric) {
+    return (d > 500) ? std::pair{QString::number(round_distance(d / 1000)), QObject::tr("km")}
+                     : std::pair{QString::number(50 * nearbyint(d / 50)), QObject::tr("m")};
+  } else {
+    float feet = d * METER_TO_FOOT;
+    return (feet > 500) ? std::pair{QString::number(round_distance(d * METER_TO_MILE)), QObject::tr("mi")}
+                        : std::pair{QString::number(50 * nearbyint(d / 50)), QObject::tr("ft")};
```
Comment (deanlee) [inline selfdrive/ui/qt/maps/map_helpers.cc:145]: std::nearbyint

**Comment by sshane** [comment]
thanks!

**Pull request merged**